tiledb/sm/array_schema/array_schema.h

```diff
@@ -794,6 +796,17 @@ class ArraySchema {
 
 }  // namespace tiledb::sm
 
+/**
+ * Expands the input domain so that it aligns with the boundaries of the array's
+ * regular tiles (i.e., it maps the domain onto the regular tile grid) in the
+ * same way as Domain::expand_to_tiles(NDRange*), but while respecting the
+ * current domain.
+ */
```

Comment:
```suggestion
/**
 * Expands the input query domain (query_ndrange) so that it aligns with the boundaries
 *  of the array's regular tiles. (i.e., it maps the domain onto the regular tile grid) in the
 * same way as Domain::expand_to_tiles(NDRange*), but while respecting the
 * current domain.
 */
```

Reply:
We should probably also rename `Domain::expand_to_tiles`, maybe to `Domain::_expand_to_tiles` or `Domain::expand_to_tiles_without_current_domain`, and/or put a comment on it, saying, do not call this, but call `expand_tiles_respecting_current_domain` instead.

Reply:
I wouldn't recommend renaming `Domain::expand_to_tiles` since it is still being used.